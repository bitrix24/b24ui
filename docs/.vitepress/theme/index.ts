/**
 * This is custom theme
 * We change some css
 * @see node_modules/vitepress/dist/client/theme-default/index.js
 */
import { ref, watch, toValue } from 'vue'
import { createRouter, createMemoryHistory } from 'vue-router'
import bitrix24UIPlugin from '@bitrix24/b24ui-nuxt/vue-plugin'
import Layout from './components/ui/Layout.vue'
import ComponentShowExample from './components/ui/ComponentShowExample.vue'
import ComponentProps from './components/ui/ComponentProps.vue'
import ComponentSlots from './components/ui/ComponentSlots.vue'
import ComponentEmits from './components/ui/ComponentEmits.vue'
import Description from './components/ui/Description.vue'
import Placeholder from './components/ui/Placeholder.vue'
import VPBadge from 'vitepress/dist/client/theme-default/components/VPBadge.vue'
import 'vitepress/dist/client/theme-default/styles/fonts.css'
import './tailwind.css'
// region ory theme ////
import './styles/components/custom-block.css'
import './styles/components/vp-code.css'
import './styles/components/vp-code-group.css'
import './styles/components/vp-doc.css'
import './styles/components/vp-sponsor.css'
// endregion ////
import './theme.css'

globalThis.useFetch = async (url: string, options: RequestInit & { params?: any, transform?: (data) => any } = {}) => {
  const data = ref()
  const status = ref('idle')
  async function _fetch() {
    status.value = 'loading'
    try {
      const params = options.params
        ? Object.fromEntries(
            Object.entries(options.params).map(([key, value]) => [key, toValue(value)])
          )
        : {}

      const queryString = new URLSearchParams(params as any).toString()
      const fullUrl = queryString ? `${url}?${queryString}` : url

      data.value = await fetch(fullUrl, options).then(r => r.json()).then(r => options.transform ? options.transform(r) : r)
      status.value = 'success'
    } catch (error) {
      console.error(error)
      status.value = 'error'
    }
  }
  _fetch()

  if (options?.params?.q) {
    watch(
      options.params.q,
      () => {
        _fetch()
      },
      { deep: true }
    )
  }

  return Promise.resolve({
    data,
    status
  })
}

export default {
  Layout,
  enhanceApp: async ({ app }) => {
    // region components ////
    app.component('Badge', VPBadge)
      .component('Description', Description)
      .component('ComponentShowExample', ComponentShowExample)
      .component('ComponentProps', ComponentProps)
      .component('ComponentSlots', ComponentSlots)
      .component('ComponentEmits', ComponentEmits)
      .component('Placeholder', Placeholder)
    // endregion ////

    if (!import.meta.env.SSR) {
      const router = createRouter({
        routes: [
          // { path: '', component: app }
        ],
        history: createMemoryHistory()
        // history: createWebHistory()
      })
      app.use(router)

      app.use(bitrix24UIPlugin)
    }
  }
}
