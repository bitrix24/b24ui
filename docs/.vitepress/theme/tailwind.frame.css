/*
@plugin '@bitrix24/b24style';
@plugin "@bitrix24/b24ui-nuxt";
*/
@import "tailwindcss" source("../../");
@import "@bitrix24/b24ui-nuxt";
/*
*/



