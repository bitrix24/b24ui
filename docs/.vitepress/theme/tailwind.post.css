@import "tailwindcss" source("../../../docs");
@import "@bitrix24/b24ui-nuxt";

/**
 * bxshef: Custom
 * -------------------------------------------------------------------------- */

:root
{
}

.dark
{
  --vp-c-bg: #0f172a;
  --vp-c-bg-alt: #161618;
  --vp-c-bg-elv: #202127;
  --vp-c-bg-soft: #202127;
}

.bg-grid-example {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

.dark\:bg-grid-example:is(.dark *) {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}
