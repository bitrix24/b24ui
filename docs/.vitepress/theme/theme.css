/**
 * bxshef: Custom
 * -------------------------------------------------------------------------- */

:root
{
  --sh-scrollbar-thumb: #3c3c43;
  --sh-scrollbar-background: #ebebef;
}

.dark
{
  --vp-c-bg: #0f172a;
  --vp-c-bg-alt: #161618;
  --vp-c-bg-elv: #202127;
  --vp-c-bg-soft: #202127;
  --vp-code-block-bg: #1a2234;

  --sh-scrollbar-thumb: #bdc1c6;
  --sh-scrollbar-background: #3c3c43;
}

.bg-grid-example {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

.dark\:bg-grid-example:is(.dark *) {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

iframe
{
  min-width: 100%;
  border: none;
  scrollbar-width: thin;
  background-color: transparent;
}

/** region scrollbar */
* {
  scrollbar-color: var(--sh-scrollbar-thumb) var(--sh-scrollbar-background);
  scrollbar-width: thin;
}

/* Styling the scrollbar in Chrome and Safari */
*::-webkit-scrollbar {
  width: 6px;
}

/* Set thumb color for Chrome and Safari */
*::-webkit-scrollbar-thumb {
  background-color: var(--sh-scrollbar-thumb);
  border-radius: 4px;
}

/* Set track color for Chrome and Safari */
*::-webkit-scrollbar-track {
  background-color: var(--sh-scrollbar-background);
}
/** endregion */
