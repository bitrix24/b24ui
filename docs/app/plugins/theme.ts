export default defineNuxtPlugin({
  enforce: 'post',
  setup() {
  }
})
