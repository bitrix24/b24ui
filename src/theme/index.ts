export { default as advice } from './advice'
export { default as avatar } from './avatar'
export { default as avatarGroup } from './avatar-group'
export { default as badge } from './badge'
export { default as button } from './button'
export { default as buttonGroup } from './button-group'
export { default as checkbox } from './checkbox'
export { default as chip } from './chip'
export { default as container } from './container'
export { default as countdown } from './countdown'
export { default as formField } from './form-field'
export { default as input } from './input'
export { default as kbd } from './kbd'
export { default as link } from './link'
export { default as progress } from './progress'
export { default as radioGroup } from './radio-group'
export { default as range } from './range'
export { default as separator } from './separator'
export { default as skeleton } from './skeleton'
export { default as switch } from './switch'
export { default as textarea } from './textarea'
export { default as tooltip } from './tooltip'
