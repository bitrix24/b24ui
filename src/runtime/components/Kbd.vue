<script lang="ts">
import type { AppConfig } from '@nuxt/schema'
import theme from '#build/b24ui/kbd'
import type { KbdKey } from '../composables/useKbd'
import type { ComponentConfig } from '../types/utils'

type Kbd = ComponentConfig<typeof theme, AppConfig, 'kbd'>

/**
 * @remove depth
 */
export interface KbdProps {
  /**
   * The element or component this component should render as.
   * @defaultValue 'kbd'
   */
  as?: any
  value?: KbdKey | string
  /**
   * @defaultValue 'default'
   */
  accent?: Kbd['variants']['accent']
  /**
   * @defaultValue 'md'
   */
  size?: Kbd['variants']['size']
  class?: any
}

export interface KbdSlots {
  default(props?: {}): any
}
</script>

<script setup lang="ts">
import { computed } from 'vue'
import { Primitive } from 'reka-ui'
import { useAppConfig } from '#imports'
import { useKbd } from '../composables/useKbd'
import { tv } from '../utils/tv'

const props = withDefaults(defineProps<KbdProps>(), {
  as: 'kbd',
  accent: 'default'
})
defineSlots<KbdSlots>()

const { getKbdKey } = useKbd()

const appConfig = useAppConfig() as Kbd['AppConfig']

const b24ui = computed(() => tv({ extend: tv(theme), ...(appConfig.b24ui?.kbd || {}) })({
  accent: props.accent,
  size: props.size
}))
</script>

<template>
  <Primitive :as="as" :class="b24ui.base({ class: props.class })">
    <slot>
      {{ getKbdKey(value) }}
    </slot>
  </Primitive>
</template>
