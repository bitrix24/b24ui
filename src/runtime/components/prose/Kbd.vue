<script lang="ts">
import type { AppConfig } from '@nuxt/schema'
import theme from '#build/b24ui/prose/kbd'
import type { ComponentConfig } from '../../types/tv'

type ProseKbd = ComponentConfig<typeof theme, AppConfig, 'kbd', 'b24ui.prose'>

export interface ProseKbdProps {
  value: string
  class?: any
}
</script>

<script setup lang="ts">
import { computed } from 'vue'
import { useAppConfig } from '#imports'
import { tv } from '../../utils/tv'
import B24Kbd from '../Kbd.vue'

const props = defineProps<ProseKbdProps>()

const appConfig = useAppConfig() as ProseKbd['AppConfig']

const b24ui = computed(() => tv({ extend: tv(theme), ...(appConfig.b24ui?.prose?.kbd || {}) }))
</script>

<template>
  <B24Kbd :value="value" :class="b24ui({ class: props.class })" />
</template>
