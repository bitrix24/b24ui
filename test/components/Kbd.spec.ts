import { describe, it, expect } from 'vitest'
import Kbd from '../../src/runtime/components/Kbd.vue'
import type { KbdProps, KbdSlots } from '../../src/runtime/components/Kbd.vue'
import ComponentRender from '../component-render'
import theme from '#build/b24ui/kbd'

describe('Kbd', () => {
  const sizes = Object.keys(theme.variants.size) as any
  // const depths = Object.keys(theme.variants.depth) as any

  it.each([
    // Props
    ['with value', { props: { value: 'K' } }],
    ...sizes.map((size: string) => [`with size ${size}`, { props: { value: 'K', size } }]),
    // ...depths.map((depth: string) => [`with depth ${depth}`, { props: { value: 'K', depth } }]),
    ['with as', { props: { value: 'K', as: 'span' } }],
    ['with class', { props: { value: 'K', class: 'font-bold' } }],
    // Slots
    ['with default slot', { slots: { default: () => 'Default slot' } }]
  ])('renders %s correctly', async (nameOrHtml: string, options: { props?: KbdProps, slots?: Partial<KbdSlots> }) => {
    const html = await ComponentRender(nameOrHtml, options, Kbd)
    expect(html).toMatchSnapshot()
  })
})
