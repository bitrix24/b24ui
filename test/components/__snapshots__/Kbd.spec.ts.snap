// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Kbd > renders with accent accent correctly 1`] = `"<kbd class="inline-flex items-center justify-center px-1 rounded-(--ui-border-radius-2xs) font-(--ui-font-weight-normal) font-[family-name:var(--ui-font-family-system-mono)] uppercase border-(length:--b24ui-border-width) border-(--b24ui-border-color) text-(--b24ui-color) bg-(--b24ui-background) style-outline-accent-1 h-[24px] min-w-[24px] text-(length:--ui-font-size-md)/(--ui-font-line-height-reset)">K</kbd>"`;

exports[`Kbd > renders with accent default correctly 1`] = `"<kbd class="inline-flex items-center justify-center px-1 rounded-(--ui-border-radius-2xs) font-(--ui-font-weight-normal) font-[family-name:var(--ui-font-family-system-mono)] uppercase border-(length:--b24ui-border-width) border-(--b24ui-border-color) text-(--b24ui-color) bg-(--b24ui-background) style-outline h-[24px] min-w-[24px] text-(length:--ui-font-size-md)/(--ui-font-line-height-reset)">K</kbd>"`;

exports[`Kbd > renders with accent less correctly 1`] = `"<kbd class="inline-flex items-center justify-center px-1 rounded-(--ui-border-radius-2xs) font-(--ui-font-weight-normal) font-[family-name:var(--ui-font-family-system-mono)] uppercase border-(length:--b24ui-border-width) border-(--b24ui-border-color) text-(--b24ui-color) bg-(--b24ui-background) style-outline-no-accent h-[24px] min-w-[24px] text-(length:--ui-font-size-md)/(--ui-font-line-height-reset)">K</kbd>"`;

exports[`Kbd > renders with as correctly 1`] = `"<span class="inline-flex items-center justify-center px-1 rounded-(--ui-border-radius-2xs) font-(--ui-font-weight-normal) font-[family-name:var(--ui-font-family-system-mono)] uppercase border-(length:--b24ui-border-width) border-(--b24ui-border-color) text-(--b24ui-color) bg-(--b24ui-background) style-outline h-[24px] min-w-[24px] text-(length:--ui-font-size-md)/(--ui-font-line-height-reset)">K</span>"`;

exports[`Kbd > renders with class correctly 1`] = `"<kbd class="inline-flex items-center justify-center px-1 rounded-(--ui-border-radius-2xs) font-[family-name:var(--ui-font-family-system-mono)] uppercase border-(length:--b24ui-border-width) border-(--b24ui-border-color) text-(--b24ui-color) bg-(--b24ui-background) style-outline h-[24px] min-w-[24px] text-(length:--ui-font-size-md)/(--ui-font-line-height-reset) font-(--ui-font-weight-bold)">K</kbd>"`;

exports[`Kbd > renders with default slot correctly 1`] = `"<kbd class="inline-flex items-center justify-center px-1 rounded-(--ui-border-radius-2xs) font-(--ui-font-weight-normal) font-[family-name:var(--ui-font-family-system-mono)] uppercase border-(length:--b24ui-border-width) border-(--b24ui-border-color) text-(--b24ui-color) bg-(--b24ui-background) style-outline h-[24px] min-w-[24px] text-(length:--ui-font-size-md)/(--ui-font-line-height-reset)">Default slot</kbd>"`;

exports[`Kbd > renders with size lg correctly 1`] = `"<kbd class="inline-flex items-center justify-center px-1 rounded-(--ui-border-radius-2xs) font-(--ui-font-weight-normal) font-[family-name:var(--ui-font-family-system-mono)] uppercase border-(length:--b24ui-border-width) border-(--b24ui-border-color) text-(--b24ui-color) bg-(--b24ui-background) style-outline h-[28px] min-w-[28px] text-(length:--ui-font-size-2xl)/(--ui-font-line-height-reset)">K</kbd>"`;

exports[`Kbd > renders with size md correctly 1`] = `"<kbd class="inline-flex items-center justify-center px-1 rounded-(--ui-border-radius-2xs) font-(--ui-font-weight-normal) font-[family-name:var(--ui-font-family-system-mono)] uppercase border-(length:--b24ui-border-width) border-(--b24ui-border-color) text-(--b24ui-color) bg-(--b24ui-background) style-outline h-[24px] min-w-[24px] text-(length:--ui-font-size-md)/(--ui-font-line-height-reset)">K</kbd>"`;

exports[`Kbd > renders with size sm correctly 1`] = `"<kbd class="inline-flex items-center justify-center px-1 rounded-(--ui-border-radius-2xs) font-(--ui-font-weight-normal) font-[family-name:var(--ui-font-family-system-mono)] uppercase border-(length:--b24ui-border-width) border-(--b24ui-border-color) text-(--b24ui-color) bg-(--b24ui-background) style-outline h-[20px] min-w-[20px] text-(length:--ui-font-size-4xs)/(--ui-font-line-height-reset)">K</kbd>"`;

exports[`Kbd > renders with value correctly 1`] = `"<kbd class="inline-flex items-center justify-center px-1 rounded-(--ui-border-radius-2xs) font-(--ui-font-weight-normal) font-[family-name:var(--ui-font-family-system-mono)] uppercase border-(length:--b24ui-border-width) border-(--b24ui-border-color) text-(--b24ui-color) bg-(--b24ui-background) style-outline h-[24px] min-w-[24px] text-(length:--ui-font-size-md)/(--ui-font-line-height-reset)">K</kbd>"`;
